### ISSUE TITLE
Inconsistency between response from VillainResource.getAllVillains and @ApiResponse with responseCode 204

### ISSUE THREAD

**Issue by omasseau** [quarkusio/quarkus-workshops#218]
Hello,

I'm continuing to progress in the workshop, and found this inconsistency :) : 

The VillainResource.getAllVillains method always returns a 200 response even if there are no villains.
This is not consistent with the defined @ApiResponse : 

![image](https://user-images.githubusercontent.com/11772429/229182732-b3124662-f707-48b2-9815-fa42c0e70f54.png)

![image](https://user-images.githubusercontent.com/11772429/229182874-6fda6ff0-cd2e-4ed9-999c-f38c4850d224.png)

I guess, the @ApiResponse should be completely removed or the code should be modified : 

```
    @Operation(summary = "Returns all the villains from the database")
    @GET
    @APIResponse(responseCode = "200", content = @Content(mediaType = APPLICATION_JSON, schema = @Schema(implementation = Villain.class, type = SchemaType.ARRAY)))
    @APIResponse(responseCode = "204", description = "No villains")
    public RestResponse<?> getAllVillains() {
        List<Villain> villains = service.findAllVillains();
        logger.debug("Total number of villains " + villains);
        
	if (!villains.isEmpty()) {
	    return RestResponse.ok(villains);
        } else {
            return RestResponse.noContent();
        }
    }
```

**Comment by omasseau** [comment]
I just noticed it is also the case in HeroResource : https://quarkus.io/quarkus-workshops/super-heroes/#_the_hero_resource

![image](https://user-images.githubusercontent.com/11772429/229289395-f7493462-2b86-4976-88ca-02dcb823b1a2.png)

Further things I noticed : 
- Sometimes the Uni is encapsulated in a Response, sometimes it is not.
It can be confusing for beginners why sometimes it is encapsulated (logically when a response code != 200 is needed), and why sometimes it is not.
Some comments in the doc about why would be nice I think :)

- Response is used instead of RestResponse

I can submit a PR. Just need to know before if : 
- You prefer to keep the possibility to return a 204 response code or prefer to return a 200 response code if there are no items.
- You prefrer to use RestResponse instead of Response or not.
